pkg/env/env.go

```diff
@@ -431,29 +420,80 @@ func (e *testEnv) execFeature(ctx context.Context, t *testing.T, featName string
 				assessName = fmt.Sprintf("Assessment-%d", i+1)
 			}
 			t.Run(assessName, func(t *testing.T) {
-				// skip assessments which matches with --skip-assessments
-				if e.cfg.SkipAssessmentRegex() != nil && e.cfg.SkipAssessmentRegex().MatchString(assess.Name()) {
-					t.Skipf(`Skipping assessment "%s": name matched`, assess.Name())
+				skipped, message := e.requireProcessing(nil, assess, i+1)
+				if skipped {
+					t.Skipf(message)
 				}
-
-				// skip assessments which does not matches with --assess
-				if e.cfg.AssessmentRegex() != nil && !e.cfg.AssessmentRegex().MatchString(assess.Name()) {
-					t.Skipf(`Skipping assessment "%s": name not matched`, assess.Name())
-				}
-				ctx = assess.Func()(ctx, t, e.cfg)
+				ctx = e.executeSteps(ctx, t, []types.Step{assess})
 			})
 		}
 
 		// teardowns run at feature-level
 		teardowns := features.GetStepsByLevel(f.Steps(), types.LevelTeardown)
-		for _, teardown := range teardowns {
-			ctx = teardown.Func()(ctx, t, e.cfg)
-		}
+		ctx = e.executeSteps(ctx, t, teardowns)
 	})
 
 	return ctx
 }
 
+// requireProcessing is used as a generic helper utility that can be used to decide if a feature or assessment
+// needs to be skipped or not. This will simplify the process of checking for these behavior during the dry-run
+// mode. This also cleans up the duplication of code we had in execFeature with regards to how the skip was being
+// detected for feature and assessment
+func (e *testEnv) requireProcessing(f types.Feature, a types.Step, i int) (skip bool, message string) {
```

Comment:
Please document what `i` is or give it a more descriptive name

Reply:
Or, instead of `i`, pass in a `defaultName` string that will be used if feature does not have a name.

Reply:
Actually, I think you should consider breaking this into two methods:

* requireFeatureProcessing()
* requireStepProcessing()